roles/servicetelemetry/tasks/pre.yml

```diff
@@ -79,6 +79,54 @@
     namespace: "{{ ansible_operator_meta.namespace }}"
   register: smartgateways_loaded
 
+- name: Get current STF object
+  k8s_info:
+    api_version: infra.watch/v1beta1
+    kind: ServiceTelemetry
+    name: "{{ ansible_operator_meta.name }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: _stf_object
+
+- name: Get community Prometheus objects
+  k8s_info:
+    api_version: monitoring.coreos.com/v1
+    kind: Prometheus
+    name: "{{ ansible_operator_meta.name }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: _community_prom_object
+
+- block:
+  - name: Apply community observability_strategy if missing on an STF object with an existing community prometheus
+    k8s:
+      definition:
+        apiVersion: infra.watch/v1beta1
+        kind: ServiceTelemetry
+        metadata:
+          name: "{{ ansible_operator_meta.name }}"
+          namespace: "{{ ansible_operator_meta.namespace }}"
+        spec:
+          observability_strategy: use_community
```

Comment:
this works because `k8s` module uses a `merge` strategy by default?

Reply:
:heavy_check_mark: